spec-tests/src/test/java/org/candlepin/spec/EntitlementCertificateSpecTest.java

```diff
@@ -0,0 +1,474 @@
+        EntitlementDTO ent = consumerApi.listEntitlements(system.getUuid()).get(0);
+        OffsetDateTime test = subscription != null ? subscription.getEndDate() : pool.getEndDate();
+        assertEquals(ent.getEndDate(), test);
+    }
+
+    @Test
+    @DisplayName("single entitlement in excess will be deleted when " +
+        "existing subscription quantity is decreased")
+    public void deleteEntWhenSubQuantityDecreased() throws Exception {
+        //  this entitlement makes the counts inconclusive
+        consumerApi.unbindAll(system.getUuid());
+        ProductDTO prod = Products.randomEng().addAttributesItem(
+            new AttributeDTO().name("multi-entitlement").value("yes"));
+        if (CandlepinMode.isHosted()) {
+            prod = hostedTestApi.createProduct(prod);
+            subscription = Subscriptions.random(owner, prod);
+            subscription = hostedTestApi.createSubscription(subscription);
+            AsyncJobStatusDTO refresh = ownerApi.refreshPools(owner.getKey(), false);
+            if (refresh != null) {
+                client.jobs().waitForJob(refresh.getId());
+            }
+            List<PoolDTO> pools = ownerApi.listOwnerPools(owner.getKey());
+            for (PoolDTO subPool : pools) {
+                if (subPool.getProductId().equals(prod.getId())) {
+                    pool = subPool;
+                }
+            }
+        }
+        else {
+            prod = ownerProductApi.createProductByOwner(owner.getKey(), prod);
+            pool = Pools.random(prod);
+            pool = ownerApi.createPool(owner.getKey(), pool);
+        }
+        consumerApi.bindPool(system.getUuid(), pool.getId(), 6);
+        assertThat(consumerApi.fetchCertificates(system.getUuid())).hasSize(1);
+
+        if (CandlepinMode.isHosted()) {
+            subscription.setQuantity(subscription.getQuantity() - 5);
+            subscription = hostedTestApi.updateSubscription(subscription.getId(), subscription);
+            AsyncJobStatusDTO refresh = ownerApi.refreshPools(owner.getKey(), false);
+            if (refresh != null) {
+                client.jobs().waitForJob(refresh.getId());
+            }
+        }
+        else {
+            pool.setQuantity(pool.getQuantity() - 5);
+            ownerApi.updatePool(owner.getKey(), pool);
+        }
+        assertThat(consumerApi.fetchCertificates(system.getUuid())).hasSize(0);
+    }
+
+    @Test
+    @DisplayName("multiple entitlement in excess will be deleted when " +
+        "existing subscription quantity is decreased")
+    public void deleteMultiEntWhenSubQuantityDecreased() throws Exception {
+        //  this entitlement makes the counts inconclusive
+        consumerApi.unbindAll(system.getUuid());
+        ProductDTO prod = Products.randomEng().addAttributesItem(
+            new AttributeDTO().name("multi-entitlement").value("yes"));
+        if (CandlepinMode.isHosted()) {
```

Comment:
Yes. A lot of those tests are only merged into one because of the old `create_pool_and_subscription` method which did everything in one and hid the details. From what I recall when I attempted to rip it out, very few tests actually *needed* the hosted test bits, and even fewer had reason to change test operation depending on the mode.

Breaking up the tests will, at the very least, make it obvious which is which, and will probably also highlight areas where the differentiation isn't necessary anymore.

Reply:
Yeah, I agree on breaking up the tests based on hosted/standalone. I had to do the same for owner product resource spec tests and it makes the test more clear on what is being tested.